### PR TITLE
Updates for migration/translation settings

### PR DESCRIPTION
**Pull request by tomas862**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop / 1.7.4.x / 1.6.1.x
| Description?  | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc.
| Type?         | bug fix / improvement / new feature
| Category?     | FO / BO / CO / IN / WS / TE
| BC breaks?    | Does it break backward compatibility? yes/no
| Deprecations? | Does it deprecate an existing feature? yes/no
| Fixed ticket? | (optional) If this PR fixes a [Forge](http://forge.prestashop.com/) ticket, please add its complete Forge URL.
| How to test?  | Please indicate how to best verify that this PR is correct.

### CODE REVIEW

**Review by tomas862 — COMMENTED** [review]
I believe there might be a better solution in the commit  
 [b972fcf](https://github.com/rokaszygmantas/PrestaShop/pull/3/commits/b972fcf2556678bfb87755b0f7c3b199626c03d3) - If there are no choices I want to display only the error message but do not display the selector itself. In this commit - its how exactly it is done but I'm open for suggestions of improvement here.

File: src/Core/Form/ChoiceProvider/NonInstalledLocalizationChoiceProvider.php [hunk 1]
```diff
@@ -26,45 +26,56 @@
...
 
     public function getChoices()
     {
-        $isEmpty = empty($this->allLocalizationChoices);
-
-        if ($isEmpty) {
-            return [];
-        }
-
+        $languages = $this->languagePackLoader->getLanguagePackList();
```
Comment (rokaszygmantas) [inline src/Core/Form/ChoiceProvider/NonInstalledLocalizationChoiceProvider.php:66]: why not make this return an `array` consistently, so you don't have to care about it being `null` later on

File: src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ */
+
+namespace PrestaShop\PrestaShop\Core\Language\Pack\Loader;
+
+use PrestaShop\PrestaShop\Core\Foundation\Version;
+
+/**
+ * Class RemoteLanguagePackLoader is responsible for retrieving language pack data from remote host
+ */
+class RemoteLanguagePackLoader implements LanguagePackLoaderInterface
```
Comment (rokaszygmantas) [inline src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php]: `final`

File: src/PrestaShopBundle/Controller/Admin/TranslationsController.php [hunk 3]
```diff
@@ -238,6 +217,41 @@ public function addUpdateLanguageAction(Request $request)
...
+            $isoCode = $data['iso_code'];
+
+            $langRepository = $this->get('prestashop.core.admin.lang.repository');
+            $locale = $langRepository->getLocaleByIsoCode($isoCode);
+
+            $themeExporter = $this->get('prestashop.translation.theme.exporter');
+            $zipFile = $themeExporter->createZipArchive($themeName, $locale, _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR);
+
+            $response = new BinaryFileResponse($zipFile);
+            $response->deleteFileAfterSend($shouldDelete = true);
```
Comment (rokaszygmantas) [inline src/PrestaShopBundle/Controller/Admin/TranslationsController.php]: `$shouldDelete = true` -> `true`

**Review by rokaszygmantas — CHANGES_REQUESTED** [review]

File: src/Core/Form/ChoiceProvider/NonInstalledLocalizationChoiceProvider.php [hunk 4]
```diff
@@ -26,45 +26,56 @@
...
         $choices = [];
-        foreach ($this->allLocalizationChoices as $key => $isoCode) {
-            $isNotInstalled = !$this->languageValidator->isInstalledByIsoCode($isoCode);
+        if (!empty($languages)) {
```
Comment (rokaszygmantas) [inline src/Core/Form/ChoiceProvider/NonInstalledLocalizationChoiceProvider.php]: if `$languages` is always an array there is no need to check if it's `!empty`

**Review by rokaszygmantas — CHANGES_REQUESTED** [review]

**Pull request merged**